hypothesis-python/src/hypothesis/extra/django/_fields.py

```diff
@@ -143,13 +194,26 @@ def _for_text(field):
         # compute intersections of the full Python regex language.
         return st.one_of(*[st.from_regex(r) for r in regexes])
     # If there are no (usable) regexes, we use a standard text strategy.
-    return st.text(
+    min_size = 1
+    if getattr(field, "blank", False) or not getattr(field, "required", True):
+        min_size = 0
+    strategy = st.text(
         alphabet=st.characters(
             blacklist_characters=u"\x00", blacklist_categories=("Cs",)
         ),
-        min_size=(0 if field.blank else 1),
+        min_size=min_size,
         max_size=field.max_length,
     )
+    if getattr(field, "required", True):
+        strategy = strategy.filter(str.strip)
```

Comment:
Turns out that this fails on Python 2, because `str` is not the text type... I can't wait for 2020!

Either `text_type.strip` (from `hypothesis.internal.compat`), or `lambda s: s.strip()` would work.

Reply:
2020 can't come soon enough for so many reasons...